channel/channel.go

```diff
@@ -143,6 +144,18 @@ func (c Channel) LatestSupportedState() (state.State, error) {
 	return c.SignedStateForTurnNum[c.latestSupportedStateTurnNum].State(), nil
 }
 
+// LatestSignedState fetches the state with the largest turn number signed by any participant
+func (c Channel) LatestSignedState() state.SignedState {
+	numStates := len(c.SignedStateForTurnNum)
+	keys := make([]uint64, 0, len(c.SignedStateForTurnNum))
+	for k := range c.SignedStateForTurnNum {
+		keys = append(keys, k)
+	}
+	sort.Slice(keys, func(i, j int) bool { return keys[i] < keys[j] })
+	latestTurn := keys[numStates-1]
+	return c.SignedStateForTurnNum[latestTurn]
+}
+
```

Comment:
Drive-by comment:

- this function may want test coverage
- this looks correct to me, but something like

```
latestTurn := -1
for k := range c.SignedStateForTurnNum {
	if k > greatest {
		latestTurn = k
	}
}
return c.SignedStateForTurnNum[latestTurn]
```
is maybe easier to read & is o(n) rather than n^2

Reply:
Good call! Addressed in 17abec6..58eccbf